### DISCUSSION TITLE
PNG images are not handled/extracted correctly

### DISCUSSION THREAD

**Discussion by syntax-surgeon** [py-pdf/pypdf#2317] [Q&A]
Hi all

I have been wanting to exract all images in a PDF file as separate image files. The process seems to be causing errors during the extraction of PNG images, however, the JPEG images seem to be working just fine. The script I am using is as follows:

```
import PyPDF2
from PIL import Image
import io

doc = PyPDF2.PdfReader("data/big_lorem_multipic.pdf")

for page in doc.pages:
    for image in page.images:
        image_bytes = io.BytesIO(image.data)
        pil_image = Image.open(image_bytes)
        pil_image.save(f"data/{image.name}")
```

The JPEG image is saved perfectly, but the PNG images are not. What seems to be the issue here? I have tried manually opening the file such as: `with open(f"data/{image.name}", "wb) as file: file.write(image.data)` and the results have been exactly the same. I can see that a related bug fix (https://github.com/py-pdf/pypdf/pull/1834) was recently made but I still cannot identify the cause of this issue.

For reference I am attaching the PDF below. Thank you
[big_lorem_multipic.pdf](https://github.com/py-pdf/pypdf/files/13487249/big_lorem_multipic.pdf)

**Comment by stefan6419846** [comment]
This probably should be reported as an issue once you migrated from the deprecated PyPDF2 to *pypdf* (simplified code):

```python3
import pypdf


doc = pypdf.PdfReader("big_lorem_multipic.pdf")

for page in doc.pages:
    for image in page.images:
        image.image.save(image.name)
```

**Reply by syntax-surgeon** [reply]
Thank you. Moved as an issue here: https://github.com/py-pdf/pypdf/issues/2318
